timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/AsyncLockService.java

```diff
@@ -39,6 +41,7 @@
     private final ScheduledExecutorService reaperExecutor;
     private final HeldLocksCollection heldLocks;
     private final ImmutableTimestampTracker immutableTsTracker;
+    private final ConcurrentMap<UUID, AsyncResult<Void>> waitRequests = Maps.newConcurrentMap();
```

Comment:
I think you meant to use the `AwaitedLocksCollection` here, which basically puts a nice bit of abstraction around _how_ we handle inflight requests.